app/drivers/zephyr/kscan_gpio_matrix.c

```diff
@@ -31,7 +31,8 @@ struct kscan_gpio_item_config {
 #define _KSCAN_GPIO_ROW_CFG_INIT(idx, n) _KSCAN_GPIO_ITEM_CFG_INIT(n, row_gpios, idx)
 #define _KSCAN_GPIO_COL_CFG_INIT(idx, n) _KSCAN_GPIO_ITEM_CFG_INIT(n, col_gpios, idx)
 
-static int kscan_gpio_config_interrupts(struct device **devices,
+COND_CODE_0(ZMK_KSCAN_MATRIX_POLLING,
```

Comment:
Ditto, this needs to have the `CONFIG_` prefix.

Also, for entire function declarations, just use a normal `#if` or `#ifdef` preprocessor check here.